[CI] AggsTimeoutIT testTerms failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #6235 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/qweil2df65ww4)
- [elasticsearch-periodic-platform-support #6235 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/nu6jtqdg3fktw)
- [elasticsearch-periodic-platform-support #6228 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/jyfwydogbo3qs)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:analytics:javaRestTest" --tests "org.elasticsearch.multiterms.AggsTimeoutIT.testTerms" -Dtests.seed=5E3D38918BF23CA5 -Dtests.locale=gl-ES -Dtests.timezone=America/Argentina/Buenos_Aires -Druntime.java=23
```

**Applicable branches:**
9.0

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.multiterms.AggsTimeoutIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testTerms),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
               nodes: an empty map
hXfvvsWCSdqk2hJ7KwB-Ug: <unexpected> but was <{transport_address=127.0.0.1:44917, ip=127.0.0.1:44917, roles=[data, data_cold, data_content, data_frozen, data_hot, data_warm, ingest, master, ml, remote_cluster_client, transform], name=test-cluster-0, host=127.0.0.1, attributes={xpack.installed=true, transform.config_version=10.0.0, testattr=test, ml.config_version=12.0.0}, tasks={hXfvvsWCSdqk2hJ7KwB-Ug:909={node=hXfvvsWCSdqk2hJ7KwB-Ug, headers={}, running_time_in_nanos=12112677869, action=indices:data/read/search, description=indices[deep], search_type[QUERY_THEN_FETCH], source[{"size":0,"aggregations":{"agg010":{"terms":{"field":"kwd009","size":10,"min_doc_count":1,"shard_min_doc_count":0,"show_term_doc_count_error":false,"order":[{"_count":"desc"},{"_key":"asc"}]},"aggregations":{"agg009":{"terms":{"field":"kwd008","size":10,"min_doc_count":1,"shard_min_doc_count":0,"show_term_doc_count_error":false
[truncated]
```

**Issue Reasons:**
- [9.0] 2 consecutive failures in test testTerms
- [9.0] 2 consecutive failures in step oraclelinux-8_platform-support-unix
- [9.0] 3 failures in test testTerms (5.6% fail rate in 54 executions)
- [9.0] 2 failures in step oraclelinux-8_platform-support-unix (100.0% fail rate in 2 executions)
- [9.0] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/x-pack/plugin/analytics/src/javaRestTest/java/org/elasticsearch/multiterms/AggsTimeoutIT.java
+++ b/x-pack/plugin/analytics/src/javaRestTest/java/org/elasticsearch/multiterms/AggsTimeoutIT.java
@@ -8,6 +8,7 @@
 package org.elasticsearch.multiterms;
 
 import org.apache.http.client.config.RequestConfig;
+import org.apache.http.util.EntityUtils;
 import org.elasticsearch.action.admin.indices.create.CreateIndexResponse;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.common.Strings;
@@ -30,6 +31,7 @@ import java.io.IOException;
 import java.net.SocketTimeoutException;
 import java.util.Locale;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 import static org.elasticsearch.test.ListMatcher.matchesList;
 import static org.elasticsearch.test.MapMatcher.assertMap;
@@ -287,14 +289,33 @@ public class AggsTimeoutIT extends ESRestTestCase {
         request.setOptions(request.getOptions().toBuilder().setRequestConfig(config.build()));
     }
 
+    /**
+     * Asserts that within a minute the _search has left the _tasks api.
+     * <p>
+     *     It'd sure be more convenient if, whenever the _search has returned
+     *     back to us the _tasks API doesn't contain the _search. But sometimes
+     *     it still does. So long as it stops <strong>eventually</strong> that's
+     *     still indicative of the interrupt code working.
+     * </p>
+     */
     private void assertNoSearchesRunning() throws Exception {
-        Request tasks = new Request("GET", "/_tasks");
-        tasks.addParameter("actions", "*search");
-        tasks.addParameter("detailed", "");
         assertBusy(() -> {
-            Map<?, ?> response = responseAsMap(client().performRequest(tasks));
-            // If there are running searches the map in `nodes` is non-empty.
-            assertMap(response, matchesMap().entry("nodes", matchesMap()));
-        });
+            Request tasks = new Request("GET", "/_tasks");
+            tasks.addParameter("actions", "*search");
+            tasks.addParameter("detailed", "");
+            assertBusy(() -> {
+                Map<?, ?> response = responseAsMap(client().performRequest(tasks));
+                // If there are running searches the map in `nodes` is non-empty.
+                if (response.isEmpty() == false) {
+                    logger.warn("search still running, hot threads:\n{}", hotThreads());
+                }
+                assertMap(response, matchesMap().entry("nodes", matchesMap()));
+            });
+        }, 1, TimeUnit.MINUTES);
+    }
+
+    private String hotThreads() throws IOException {
+        Request tasks = new Request("GET", "/_nodes/hot_threads");
+        return EntityUtils.toString(client().performRequest(tasks).getEntity());
     }
 }

